packages/cmd/src/CommandProcessor.ts

```diff
@@ -384,11 +394,24 @@ export class CommandProcessor {
         try {
             // Build the response object, base args object, and the entire array of options for this command
             // Assume that the command succeed, it will be marked otherwise under the appropriate failure conditions
-            if (params.arguments.dcd) {
+            if (this.mDaemonResponse != null) {
                 // NOTE(Kelosky): we adjust `cwd` and do not restore it, so that multiple simultaneous requests from the same
                 // directory will operate without unexpected chdir taking place.  Multiple simultaneous requests from different
                 // directories may cause unpredictable results
-                process.chdir(params.arguments.dcd as string);
+                if (this.mDaemonResponse.cwd != null) {
+                    process.chdir(this.mDaemonResponse.cwd);
+                }
+
+                // Delete environment variables that start with CLI prefix
+                for (const envVarName of Object.keys(process.env)) {
+                    if (envVarName.startsWith(`${this.mEnvVariablePrefix}_`)) {
+                        delete process.env[envVarName];
```

Comment:
Sounds a little scary to remove all CLI prefixed ENV vars from the process.
What happens when multiple users are making use of the same daemon process ?
What if the user had Zowe's prefix (`ZOWE_`) in ENV variables that affected a specific plugin behavior ?

Reply:
Mentioned this as a limitation on the Zowe CLI PR:
> The way environment variables and stdin data are handled will need to be enhanced in the future when we support multitenancy. While worth noting, this is not a regression since the old implementation also didn't support multitenancy.

Do you think the current implementation is acceptable for now? Our codebase already makes extensive use of the `process` module for "cwd", "env", and "stdin", all of which may break if multiple users make use of the same daemon process.

Reply:
I think it's fine for the case of multitenancy.
I'm curious how this may affect plugins that may depend on a ZOWE_ ENV variable that we may be deleting here 😋

Reply:
Thanks for clarifying offline.
Ok to resolve 😋